public/app/features/manage-dashboards/components/PublicDashboardListTable.tsx

```diff
@@ -1,99 +1,115 @@
 import { css } from '@emotion/css';
-import React, { useState } from 'react';
-import useAsync from 'react-use/lib/useAsync';
+import React from 'react';
+import { useWindowSize } from 'react-use';
 
 import { GrafanaTheme2 } from '@grafana/data';
-import { getBackendSrv } from '@grafana/runtime';
-import { Link, ButtonGroup, LinkButton, Icon, Tag, useStyles2 } from '@grafana/ui';
+import { selectors as e2eSelectors } from '@grafana/e2e-selectors/src';
+import { Link, ButtonGroup, LinkButton, Icon, Tag, useStyles2, Tooltip, useTheme2, Spinner } from '@grafana/ui';
+import { Page } from 'app/core/components/Page/Page';
 import { getConfig } from 'app/core/config';
 
-export interface ListPublicDashboardResponse {
-  uid: string;
-  accessToken: string;
-  dashboardUid: string;
-  title: string;
-  isEnabled: boolean;
-}
+import { contextSrv } from '../../../core/services/context_srv';
+import { AccessControlAction } from '../../../types';
+import { useGetPublicDashboardsQuery } from '../../dashboard/api/publicDashboardApi';
+import { isOrgAdmin } from '../../plugins/admin/permissions';
 
-export const LIST_PUBLIC_DASHBOARD_URL = `/api/dashboards/public-dashboards`;
-export const getPublicDashboards = async (): Promise<ListPublicDashboardResponse[]> => {
-  return getBackendSrv().get(LIST_PUBLIC_DASHBOARD_URL);
-};
+import { DeletePublicDashboardButton } from './DeletePublicDashboard/DeletePublicDashboardButton';
 
-export const viewPublicDashboardUrl = (accessToken: string): string => {
-  return `${getConfig().appUrl}public-dashboards/${accessToken}`;
-};
+export const viewPublicDashboardUrl = (accessToken: string): string =>
+  `${getConfig().appUrl}public-dashboards/${accessToken}`;
+
+export const PublicDashboardListTable = () => {
+  const { width } = useWindowSize();
+  const isMobile = width <= 480;
+  const theme = useTheme2();
+  const styles = useStyles2(() => getStyles(theme, isMobile));
 
-export const ListPublicDashboardTable = () => {
-  const styles = useStyles2(getStyles);
-  const [publicDashboards, setPublicDashboards] = useState<ListPublicDashboardResponse[]>([]);
+  const { data: publicDashboards, isLoading, isFetching } = useGetPublicDashboardsQuery();
 
-  useAsync(async () => {
-    const publicDashboards = await getPublicDashboards();
-    setPublicDashboards(publicDashboards);
-  }, [setPublicDashboards]);
+  const selectors = e2eSelectors.pages.PublicDashboards;
+  const hasWritePermissions = contextSrv.hasAccess(AccessControlAction.DashboardsPublicWrite, isOrgAdmin());
```

Comment:
@evictorero is this the proper way to check for permissions with RBAC ?

Reply:
yes, that's how we are doing with SharePublicDashboard component